eisbuk-admin/src/components/atoms/SlotCard/DurationsSection.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { useParams } from "react-router-dom";
+
+import Button from "@material-ui/core/Button";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import Typography from "@material-ui/core/Typography";
+import Box from "@material-ui/core/Box";
+
+import ArrowBackIcon from "@material-ui/icons/ArrowBack";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+import { Duration, Slot } from "eisbuk-shared";
+
+import { slotsLabels } from "@/config/appConfig";
+
+import {
+  subscribeToSlot,
+  unsubscribeFromSlot,
+} from "@/store/actions/bookingOperations";
+
+interface Props extends Slot<"id"> {
+  /**
+   * Duration of particular slot a customer is subscribed to.
+   */
+  subscribedDuration?: Duration;
+  /**
+   * Enable subscription actions only in customer view.
+   * Disable otherwise (and render UI accordingly).
+   */
+  enableSubscription: boolean;
+}
+
+/**
+ * Renders durations for slot and enables booking (if customer view).
+ * All booking (slot subscription) action handlers are kept internally and dispatched directly to store.
+ */
+const DurationsSection: React.FC<Props> = ({
+  subscribedDuration,
+  enableSubscription,
+  ...slotData
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const { secretKey } = useParams<{ secretKey?: string }>();
```

Comment:
I think the current URL scheme that embeds the secret key in the path should be kept: currently we have `/clienti/88de50d5-2b63-4652-9c86-bd91c6fe2c2f` (we should change `clienti` to `customer`) and that looks better than `/customer?secret_key=88de50d5-2b63-4652-9c86-bd91c6fe2c2f`.

Reply:
Yes...first off, I completely agree that query params are somewhat ugly and I prefer not to use them as well (if not needed).
Secondly, I think I understand your confusion here: 
If you see the current code, you'll see that we already use the exact same implementation.The beauty of `react-router` is that, other than (traditional) query params, it features "route params", so this routes `path` is passed like this `customers/:secret_key`. This way the `:secret_key` is automatically recognized as a route param. So the real route would still look like it already does.

On a slightly unrelated note: there's no rule as to where the route params should be, so if we, for instance, wanted to use a sub-path language routing we could do something like this `path="/:lang/customers/:secret_key", and `react-router` and `useParams()` hook would return an object with this interface: `{lang: string, secret_key: string}`

Reply:
Nice! I didn't know the same infrastructure can retrieve URL parameters and path components (the way I usually call "route parameters").